modules/aws/mx/variables.tf

```diff
@@ -0,0 +1,164 @@
+  default     = []
+}
+
+variable "role_arn" {
+  type        = string
+  default     = null
+  description = "IAM role to assign to the DSF MX. Keep empty if you wish to create a new role."
+}
+
+variable "mx_password" {
+  type        = string
+  description = "MX password"
+  sensitive   = true
+  validation {
+    # Check that the password is at least 8 characters long
+    condition     = length(var.mx_password) >= 7 && length(var.mx_password) <= 14
+    error_message = "Password must be 7-14 characters long"
+  }
+
+  validation {
+    # Check that the password contains at least one uppercase letter
+    condition     = can(regex("[A-Z]", var.mx_password))
+    error_message = "Password must contain at least one uppercase letter"
+  }
+
+  validation {
+    # Check that the password contains at least one lowercase letter
+    condition     = can(regex("[a-z]", var.mx_password))
+    error_message = "Password must contain at least one lowercase letter"
+  }
+
+  validation {
+    # Check that the password contains at least one digit
+    condition     = can(regex("\\d", var.mx_password))
+    error_message = "Password must contain at least one digit"
+  }
+
+  validation {
+    # Check that the password contains at least one special character
+    condition     = can(regex("[*+=#%^:/~.,\\[\\]_]", var.mx_password))
+    error_message = "Password must contain at least one of the following special character - *+=#%^:/~.,[]_"
+  }
+}
+
+variable "secure_password" {
+  type        = string
+  description = "The password used for communication between the Management Server and the Gateway"
+  sensitive   = true
+  validation {
+    # Check that the password is at least 8 characters long
+    condition     = length(var.secure_password) >= 7 && length(var.secure_password) <= 14
+    error_message = "Password must be 7-14 characters long"
+  }
+
+  validation {
+    # Check that the password contains at least one uppercase letter
+    condition     = can(regex("[A-Z]", var.secure_password))
+    error_message = "Password must contain at least one uppercase letter"
+  }
+
+  validation {
+    # Check that the password contains at least one lowercase letter
+    condition     = can(regex("[a-z]", var.secure_password))
+    error_message = "Password must contain at least one lowercase letter"
+  }
+
+  validation {
+    # Check that the password contains at least one digit
+    condition     = can(regex("\\d", var.secure_password))
+    error_message = "Password must contain at least one digit"
+  }
+
+  validation {
+    # Check that the password contains at least one special character
+    condition     = can(regex("[*+=#%^:/~.,\\[\\]_]", var.secure_password))
+    error_message = "Password must contain at least one of the following special character - *+=#%^:/~.,[]_"
+  }
+}
+
+variable "license_file" {
+  type        = string
+  description = "DAM license file path. Make sure this license is valid before deploying DAM"
```

Comment:
Yes. Around 30 minutes.

Reply:
Then maybe mention this in the message. For example: Make sure this license is valid before deploying DAM otherwise this will result in an invalid deployment and loss of time.